torch/csrc/jit/serialization/unpickler.cpp

```diff
@@ -868,6 +868,23 @@ void Unpickler::rebuildTensor(bool quantized) {
     impl->set_storage_offset(storage_offset);
     impl->set_sizes_and_strides(size, stride);
     result = autograd::make_variable(result, requires_grad);
+
+    // Handle if math_bits were pickled.
+    // See `args` of _reduce_ex_internal
+    // for a regular tensor (final else case).
+    // NOTE: `math_bits` is the 7th arg.
+    // NOTE: This is only meant for regular tensor and not quantized!
+    if (!quantized) {
+      bool has_math_bits = elements.size() == 7;
```

Comment:
seems a bit weird specializing on the args size ...

Reply:
I think the idea is that ideally any new meta-data should be added to this dictionary. I wonder if renaming math_bits to metadata is better?

Ref: https://github.com/pytorch/pytorch/pull/88182#issuecomment-1304265266
Ref: https://github.com/pytorch/pytorch/pull/88182#discussion_r1014395571

Reply:
Ah got it. yeah adding some comments and the rename would make it clearer